hw/misc/hermes.c

```diff
@@ -1102,6 +1148,30 @@ static void pci_hermes_realize(PCIDevice *pdev, Error **errp)
     for (i = 0; i < C2H_CHANNELS; i++) {
         hermes_bar2_init_dir_reg(&hermes->bar2.c2h[i]);
     }
+
+    for (i = 0; i < UBPF_ENGINES; i++) {
+        hermes->ubpf[i].engine = ubpf_create();
+        hermes->ubpf[i].no = i;
+        hermes->ubpf[i].entry = &hermes_bpf_thread;
+
+        sprintf(hermes->ubpf[i].name, "ubpf-%u", i);
+
+        qemu_mutex_init(&hermes->ubpf[i].bpf_mutex);
+        qemu_cond_init(&hermes->ubpf[i].bpf_cond);
+
+        qemu_thread_create(&hermes->ubpf[i].bpf_thread,
+            hermes->ubpf[i].name,
+            hermes->ubpf[i].entry,
+            &hermes->ubpf[i],
+            QEMU_THREAD_JOINABLE);
+
+        if (!hermes->ubpf[i].engine) {
+            error_propagate(errp, NULL);
```

Comment:
The second argument can't be NULL, doing that effectively bypasses error_propagate (and error_propagate should terminate the program)

You will need to create a new `Error`, set it with `error_setg()`and pass it to error_propagate.